scripts/compile_java_test_and_verify.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+mvn -B $EXTRA_FLAGS clean compile test | tee build.log
```

Comment:
Is there a reason you didn't use parameters to the script, rather than environment variables? It would seem to me to be more common to use parameters - however I am not strongly opinionated here